### PR TITLE
Implemented setSystemTime

### PR DESCRIPTION
**Pull request by nexx512**
I implemented `setSystemTime` using the `strptime` date string parser. This parser is part of the posix standard but requires the `_XOPEN_SOURCE` to be defined.
This leads to issues with `bzero` and and some other deprecated functions. I decided to remove deprecated functions and turn back to standard ANSI-C functions.

I also chose to use another format for the getSystemTime command. This might have an impact to already existing applications. But I would consider command symmetry for getSystemTime and setSystemTime more important. I also would prefer an output of getSystemTime that is system locale specific and can be parsed by `strptime`.
I thought about useind ISO date formats but that would cause problemd encoding the weekdays. So I stuck to `strftime "%c"`.

### CODE REVIEW

File: src/unit.c [hunk 1]
```diff
@@ -166,78 +165,65 @@ int setCycleTime(char *input, char *sendBuf)
...
+    t.tm_sec = bcd2dec(recv[7]);
 
-    sprintf(result, "%s,%02X.%02X.%02X%02X %02x:%02X:%02X", day,
-            recv[3], recv[2], recv[0], recv[1],
-            recv[5], recv[6], recv[7]);
+    // This might break existing applications. But changing the string to some custom format
+    // that is not recognized by strptime for parsing dates has a symmetry impact that the
+    // format for getTime is different from the format required by setTime.
+    // I would consider the command symmetry more important for the future.
+    strftime(result, 48, "%c", &t);
```
Comment (reitermarkus) [inline src/unit.c]: I'd prefer this to use [ISO 8601](https://de.wikipedia.org/wiki/ISO_8601) instead of some locale specific format.

**Comment by nexx512** [comment]
Changed to ISO format. For now the timezones are ignored though.

**Comment by nexx512** [comment]
Are there any chances that this can make it into master? If not, what needs to be changed to meet you coding guidelines?

**Comment by ThomasCr** [comment]
pls apply this PR :)

**Comment by speters** [comment]
Could you sort out the build problems for the Cygwin build of your pull request?

Maybe it is as simple as doing 
```
#ifdef __CYGWIN__
#define __tm_gmtoff tm_gmtoff
#define __tm_zone tm_zone
#endif
```
But please check before doing so. I only guessed this from the Appveyor error log.

**Pull request merged**